packages/mcp/src/server/tools/token-service.ts

```diff
@@ -0,0 +1,156 @@
+import { exec } from "child_process";
+import { promisify } from "util";
+
+const execAsync = promisify(exec);
+
+interface TokenData {
+  [theme: string]: {
+    [tokenName: string]: string;
+  };
+}
+
+interface TokenInfo {
+  name: string;
+  value: string;
+  theme: string;
+  category: string;
+}
+
+export class TokenService {
+  private static readonly UNPKG_URL = "https://unpkg.com/@vibe/style@latest/dist/colors.json";
+  private static cachedTokens: TokenInfo[] | null = null;
+
+  static async getTokens(): Promise<TokenInfo[]> {
+    if (this.cachedTokens) {
+      console.error("[Vibe MCP] Using cached tokens");
+      return this.cachedTokens;
+    }
+
+    try {
+      console.error("[Vibe MCP] Fetching tokens from unpkg...");
+      const tokens = await this.fetchFromUnpkg();
+      console.error(`[Vibe MCP] Fetched ${tokens.length} tokens, caching for session`);
+      this.cachedTokens = tokens;
+      return tokens;
+    } catch (error) {
+      console.error("[Vibe MCP] Failed to fetch tokens:", error);
+      console.error("[Vibe MCP] Returning empty array as fallback");
+      return [];
+    }
+  }
+
+  private static async fetchFromUnpkg(): Promise<TokenInfo[]> {
+    return await this.fetchWithRetry();
+  }
+
+  private static async fetchWithRetry(): Promise<TokenInfo[]> {
+    // Try native fetch first
+    try {
+      const response = await fetch(this.UNPKG_URL);
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+      const tokensData: TokenData = await response.json();
+      return this.processTokensData(tokensData);
+    } catch (fetchError) {
+      console.error("[Vibe MCP] Fetch failed (likely corporate proxy/SSL), trying curl fallback...");
+
+      // Fallback to curl
+      try {
+        const { stdout } = await execAsync(`curl -s "${this.UNPKG_URL}"`);
+        const tokensData: TokenData = JSON.parse(stdout);
+        return this.processTokensData(tokensData);
+      } catch (curlError) {
+        console.error(`[Vibe MCP] Curl fallback also failed: ${curlError}`);
+        throw fetchError; // Re-throw the original fetch error
+      }
+    }
+  }
+
+  private static processTokensData(tokensData: TokenData): TokenInfo[] {
+    const tokens: TokenInfo[] = [];
+
+    for (const [theme, themeTokens] of Object.entries(tokensData)) {
+      for (const [tokenName, tokenValue] of Object.entries(themeTokens)) {
+        tokens.push({
+          name: tokenName,
+          value: tokenValue,
+          theme,
+          category: this.categorizeToken(tokenName)
+        });
+      }
+    }
+
+    return tokens;
+  }
+
+  private static categorizeToken(tokenName: string): string {
```

Comment:
that's a good one!

Reply:
But there are some conflicts. 
So maybe you need to aggregate, not returning immediately.
So category can be multiple value